### PR TITLE
update docs for provisioning policy matching

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14528

### CODE REVIEW

**Review by sriraagworks — APPROVED** [review]
Looks good .

File: docs/modules/reference/pages/provisioning/policies.adoc [hunk 1]
```diff
@@ -10,6 +10,19 @@ For information on how to extend the functionality of these policies, see the xr
...
 
 The matchBehavior action is a boolean control that determines how the optional parameters will be evaluated.
-Setting this parameter’s value to `ALL_PARAMETERS` causes Provisiond to evaluate each optional parameter with boolean AND logic and the value `ANY_PARAMETERS` causes OR logic to be applied.
+
+ALL_PARAMETERS::
+    All defined parameters must match (AND Boolean logic) for the policy to apply.
+ANY_PARAMETERS::
+    Any defined parameters can match (OR Boolean logic) for the policy to apply.
+NO_PARAMETERS::
+    No defined parameters can match (opposite of ANY_PARAMETERS) for the policy to apply.
```
Comment (emarsh1995) [inline docs/modules/reference/pages/provisioning/policies.adoc:19]: Probably a nitpick, but I think "Boolean \<operator\> logic" would be a bit clearer.

```suggestion
ALL_PARAMETERS::
    The policy applies when all defined parameters match (Boolean AND logic).
ANY_PARAMETERS::
    The policy applies when at least one set of defined parameters matches (Boolean OR logic).
NO_PARAMETERS::
    The policy applies when no defined parameters match (opposite of ANY_PARAMETERS).
```

File: docs/modules/reference/pages/provisioning/policies.adoc [hunk 2]
```diff
@@ -10,6 +10,19 @@ For information on how to extend the functionality of these policies, see the xr
...
 
 String values are assumed to be a substring match, unless the parameter is prefixed with `~` to indicate the policy should evaluate as a regular expression.
```
Comment (emarsh1995) [inline docs/modules/reference/pages/provisioning/policies.adoc:21]: In this case, are we saying that any defined String parameters are looking for a substring match in other parameters by default? If so, I'd suggest the following:

```suggestion
String parameters are set to search for a substring match by default. If you want the policy to evaluate a String as a regular expression, prefix it with `~`.
```

File: docs/modules/reference/pages/provisioning/policies.adoc [hunk 3]
```diff
@@ -10,6 +10,19 @@ For information on how to extend the functionality of these policies, see the xr
...
+
+== Policy Ordering
```
Comment (emarsh1995) [inline docs/modules/reference/pages/provisioning/policies.adoc:23]:
```suggestion
== Policy ordering
```

File: docs/modules/reference/pages/provisioning/policies.adoc [hunk 4]
```diff
@@ -10,6 +10,19 @@ For information on how to extend the functionality of these policies, see the xr
...
+
+The order of matching is non-trivial.
+For negative actions (for example, DO_NO_PERSIST), evaluation stops after the first match.
+For positive actions, evaluation will continue on to check subsequent policies in the order they are listed.
+After all policies are checked, if two or more policies match, the last matched action will take precedence.
```
Comment (emarsh1995) [inline docs/modules/reference/pages/provisioning/policies.adoc:28]:
```suggestion
The order in which parameters are matched is important.
For negative checks (for example, DO_NOT_PERSIST), evaluation stops after the first match found.
For positive checks, evaluation will check all policies in listed order.
After all policies are checked the system determines if a match was found.
If a match is found, the action is run.
If multiple matches are found, they are run in reverse chronological order. 
```

File: docs/modules/reference/pages/provisioning/policies.adoc [hunk 5]
```diff
@@ -10,6 +10,19 @@ For information on how to extend the functionality of these policies, see the xr
 == Common policy parameters
 
 The matchBehavior action is a boolean control that determines how the optional parameters will be evaluated.
```
Comment (emarsh1995) [inline docs/modules/reference/pages/provisioning/policies.adoc:12]:
```suggestion
The matchBehavior action is a Boolean control that determines how optional parameters will be evaluated.
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]

File: docs/modules/reference/pages/provisioning/policies.adoc (hunk 2 above)
Comment (mmahacek) [reply docs/modules/reference/pages/provisioning/policies.adoc:21]:
```suggestion
String parameters are set to search for a substring match by default.
If you want the policy to evaluate a parameter against a regular expression, prefix the value with `~`.
```

**Review by Bonrob2 — APPROVED** [review]
LGTM when update about string values is committed.

**Review by emarsh1995 — APPROVED** [review]
LGTM! Thanks! 😄

**Pull request merged**